### PR TITLE
warning about undocd api endpoints

### PR DESCRIPTION
**Pull request by mystiberry-okta**
## Description:
- Warn people not to use undocumented API endpoints

### Resolves:

* [OKTA-92604](https://oktainc.atlassian.net/browse/OKTA-92604)

### CODE REVIEW

**Comment by mystiberry-okta** [comment]
@tthompson-okta Can you review this? A blast from the past!
undocd-api

**Comment by mystiberry-okta** [comment]
The last paragraph is the new content.

File: _source/_docs/api/getting_started/design_principles.md [hunk 1]
```diff
@@ -7,7 +7,11 @@ redirect_from: "/docs/getting_started/design_principles.html"
...
+The Okta API is a versioned API.  Okta reserves the right to add new parameters, properties, or resources to the API without advance notice.
+These updates are considered **non-breaking** and the compatibility rules below should be followed to ensure your application does not break.
+Breaking changes such as removing or renaming an attribute will be released as a new version of the API.  Okta will provide a migration path for new versions of APIs and will communicate timelines for end-of-life when deprecating APIs.
+
+From time to time, Okta may expose an API endpoint without documenting it. All undocumented endpoints should be considered experimental, subject to change without notice, and not covered by any agreements.
```
Comment (tthompson-okta) [inline _source/_docs/api/getting_started/design_principles.md]: I would word this slightly different: `Do not consume Okta APIs that are not documented on this site. All undocumented endpoints should be considered private/internal, subject to change without notice, and not covered by any agreements`.

Comment (mystiberry-okta) [inline _source/_docs/api/getting_started/design_principles.md]: will do!

**Pull request merged**